Code/Framework/AzToolsFramework/AzToolsFramework/AssetEditor/AssetEditorWidget.cpp

```diff
@@ -306,306 +218,91 @@ namespace AzToolsFramework
-
-                auto assetHandler = const_cast<AZ::Data::AssetHandler*>(AZ::Data::AssetManager::Instance().GetHandler(asset.GetType()));
-                if (assetHandler->SaveAssetData(asset, &byteStream))
+                if (!tab->IsDirty())
                 {
-                    AZ::IO::FileIOStream fileStream(targetFilePath.c_str(), AZ::IO::OpenMode::ModeWrite);
-                    if (!fileStream.IsOpen())
-                    {
-                        AZ_Warning(
-                            "Asset Editor Widget", false, "Could not open file for writing - invalid path (%s).", targetFilePath.c_str());
-                        return false;
-                    }
-
-                    AzFramework::ApplicationRequests::Bus::Broadcast(
-                        &AzFramework::ApplicationRequests::NormalizePathKeepCase, targetFilePath);
-                    m_expectedAddedAssetPath = targetFilePath;
-
-                    SourceControlCommandBus::Broadcast(
-                        &SourceControlCommandBus::Events::RequestEdit,
-                        targetFilePath.c_str(),
-                        true,
-                        [](bool, const SourceControlFileInfo&)
-                        {
-                        });
-
-                    fileStream.Write(byteBuffer.size(), byteBuffer.data());
-
-                    AZStd::string watchFolder;
-                    AZ::Data::AssetInfo assetInfo;
-                    bool sourceInfoFound{};
-                    AzToolsFramework::AssetSystemRequestBus::BroadcastResult(
-                        sourceInfoFound,
-                        &AzToolsFramework::AssetSystemRequestBus::Events::GetSourceInfoBySourcePath,
-                        targetFilePath.c_str(),
-                        assetInfo,
-                        watchFolder);
-
-                    if (sourceInfoFound)
-                    {
-                        m_sourceAssetId = assetInfo.m_assetId;
-                    }
-
-                    AZStd::string fileName = targetFilePath;
-
-                    if (AzFramework::StringFunc::Path::Normalize(fileName))
-                    {
-                        AzFramework::StringFunc::Path::StripPath(fileName);
-                        m_currentAsset = fileName.c_str();
-
-                        m_userSettings->m_lastSavePath = targetFilePath;
-                        AZStd::size_t findIndex = m_userSettings->m_lastSavePath.find(fileName.c_str());
-
-                        if (findIndex != AZStd::string::npos)
-                        {
-                            m_userSettings->m_lastSavePath = m_userSettings->m_lastSavePath.substr(0, findIndex);
-                        }
-                    }
-
-                    m_dirty = false;
-
-                    AddRecentPath(targetFilePath);
-
-                    SetStatusText(Status::assetCreated);
-
-                    return true;
+                    continue;
```

Comment:
Should these two checks also be followed by an else statement? I'm not sure I see the purpose of having them